### PR TITLE
[FOLSPRINGB-97] 1. upgrade spring boot 2 to spring boot 3 2. related …

### PR DESCRIPTION
**Pull request by Arghya22**
https://issues.folio.org/browse/FOLSPRINGB-97

##Purpose 
1. Upgrade postgresql and snakeyaml 

##Approach 
1. As per description upgrade spring boot 2 -> 3 
2. In order to upgrade to spring boot version 3 , additional changes added for compatibility

### CODE REVIEW

File: mod-spring-petstore/src/main/java/org/folio/petstore/client/config/PetStoreFeignClientConfig.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package org.folio.petstore.client.config;
 
-import feign.RequestInterceptor;
-import org.folio.spring.FolioExecutionContext;
+import feign.Client;
+import feign.okhttp.OkHttpClient;
 import org.springframework.context.annotation.Bean;
 
 public class PetStoreFeignClientConfig {
-  @Bean
```
Comment (ncovercash) [inline mod-spring-petstore/src/main/java/org/folio/petstore/client/config/PetStoreFeignClientConfig.java:8]: Why the change in indentation?

**Review by ncovercash — CHANGES_REQUESTED** [review]
Everything else looks good!

File: mod-spring-petstore/src/main/java/org/folio/petstore/client/config/PetStoreFeignClientConfig.java (hunk 1 above)
Comment (Arghya22) [reply mod-spring-petstore/src/main/java/org/folio/petstore/client/config/PetStoreFeignClientConfig.java:8]: fixed

**Review by ncovercash — APPROVED** [review]
LGTM

**Review by pavankumar181 — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**